### ISSUE TITLE
HandlebarsEngineAdapter outputs warning on property reference

### ISSUE THREAD

**Issue by jhannes** [OpenAPITools/openapi-generator#7881]
https://github.com/OpenAPITools/openapi-generator/blob/b3bc926b1d48840903a1293b0c74c49e9391966c/modules/openapi-generator/src/main/java/org/openapitools/codegen/templating/HandlebarsEngineAdapter.java#L77

When using the HandlebarsEngineAdapter, expressions such as `{{description}}` in a template file gives the warning message "`[WARNING] Unregistered helper name 'with', processing template:`". Here is an example of the error message

```
    /**
     * {{{description}}}
     */
[WARNING] Unregistered helper name 'with', processing template:
```

I'm not familiar enough with the jknack Handlebars implementation to know whether this is a real issue or not. But if my understanding is correct, the callback registered with `registerHelperMissing` should probably not log if `options.helperName.equals("with")`.

**Comment by spacether** [comment]
With is a handlebars block control word, should your code be this?
```
{{#with someObject}}
...
{{/with}}
```

**Comment by jhannes** [comment]
@spacether The change works, but it makes the conversion from Mustache to Handlebars more time-consuming. Isn't e.g. {{description}} valid Handlebars syntax?

**Comment by spacether** [comment]
Yes description is valid and description must exist as a property in the object that you are in. Where is the with in your template? That is throwing an error, right?

**Comment by spacether** [comment]
Have you seen this comment I left:
https://github.com/OpenAPITools/openapi-generator/issues/510#issuecomment-938062328
In it I describe how to bulk convert mustache to handlebars templates.
Does that help you?
